File: leetcode/760. Find Anagram Mappings/760.cpp
// https://helloacm.com/the-algorithm-to-find-anagram-mappings-between-two-arrays/
// https://leetcode.com/problems/find-anagram-mappings/

class Solution {
public:
    vector<int> anagramMappings(vector<int>& A, vector<int>& B) {
        unordered_map<int, vector<int>> table;
        for (int i = 0; i < B.size(); ++ i) {
            if (table.find(B[i]) == table.end()) {
                table[B[i]] = {i};
            } else {
                table[B[i]].push_back(i); // add duplicate index
            }
        }
        vector<int> r;
        for (int i = 0; i < A.size(); ++ i) {
            r.push_back(table[A[i]].back());  // pick a mapping index
            table[A[i]].pop_back();  // remove it from the queue
        }
        return r;
    }
};

File: leetcode/760. Find Anagram Mappings/760.java
// https://helloacm.com/the-algorithm-to-find-anagram-mappings-between-two-arrays/
// https://leetcode.com/problems/find-anagram-mappings/

class Solution {
    public int[] anagramMappings(int[] A, int[] B) {
        Map<Integer, Integer> D = new HashMap();
        for (int i = 0; i < B.length; ++i) {
            D.put(B[i], i);
        }
        int[] ans = new int[A.length];
        int t = 0;
        for (int x: A) {
            ans[t++] = D.get(x);
        }
        return ans;
    }
}
